include/field3d.hxx

```diff
@@ -228,30 +230,30 @@ class Field3D : public Field, public FieldData {
   
   /// Check if this field has yup and ydown fields
   bool hasYupYdown() const {
-    return (yup_field != nullptr) && (ydown_field != nullptr);
+    return !yup_fields.empty() and !ydown_fields.empty();
   }
 
   /// Return reference to yup field
-  Field3D& yup() { 
-    ASSERT2(yup_field != nullptr); // Check for communicate
-    return *yup_field; 
+  Field3D &yup(std::vector<Field3D>::size_type index = 0) {
+    ASSERT2(index < yup_fields.size()); // Check for communicate
```

Comment:
I don't think this comment is correct (or if it is it's not super clear what it really means)?

Reply:
Oops, I edited the code without touching the comment, a classic mistake!